hudi-aws/src/main/java/org/apache/hudi/aws/sync/AWSGlueCatalogSyncClient.java

```diff
@@ -536,11 +539,30 @@ public void updateTableSchema(String tableName, MessageType newSchema) {
           .build();
 
       awsGlue.updateTable(request).get();
+      // glue needs partition schema cascading only when columns get updated
+      // TODO: skip cascading when new fields in structs are added to the schema in last position
+      boolean cascade = config.getSplitStrings(META_SYNC_PARTITION_FIELDS).size() > 0 && !schemaDiff.getUpdateColumnTypes().isEmpty();
+      if (cascade) {
+        LOG.info("Cascading column changes to partitions");
+        List<String> allPartitions = getAllPartitions(tableName).stream()
+            .map(partition -> getStringFromPartition(table.partitionKeys(), partition.getValues()))
+            .collect(Collectors.toList());
+        updatePartitionsToTable(tableName, allPartitions);
+      }
+      awsGlue.updateTable(request).get();
     } catch (Exception e) {
       throw new HoodieGlueSyncException("Fail to update definition for table " + tableId(databaseName, tableName), e);
     }
   }
 
+  private String getStringFromPartition(List<Column> partitionKeys, List<String> values) {
+    ArrayList<String> partitionValues = new ArrayList<>();
```

Comment:
this one assume both partitions and values are same length and same order.
glad if you have any better implem for this